### PR TITLE
Prometheus: Fix heatmap query format

### LINKED ISSUE #71017: Dashboard: Problem showing distributions from heatmap
### 
# What went wrong?

**What happened**:

Instead of seeing my panel I get an error since the upgrade to 10.0.0 (persists with 10.0.1).

> Prometheus heatmap transform error: data should be a time series

The panel is a bar gauge and the query looks like this

Screenshot2023-07-03_10-35-20

**What did you expect to happen**:

Instead I would expect to see a graph like this from my Prometheus histogram metric.

screenshot2021-03-25_14-56-22

### 
# How do we reproduce it?

- Create a panel of type bar gauge 
- Change Format to Heatmap
- See error from above

### 
# What Grafana version are you using?

Grafana v10.0.1 (5a30620b85)

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

A package manager (APT, YUM, BREW, etc.)

### User's OS?

Debian 11

### User's Browser?

Firefox 114.0.2 (64-Bit)

### Is this a Regression?

Yes

### Are Datasources involved?

Prometheus

### Anything else to add?

_No response_

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Fixing issue in prometheus frontend which was not data plane compatible, and was causing the code to throw errors when looking for the old Field name of "Value", which was changed in https://github.com/grafana/grafana/pull/62694.

**Why do we need this feature?**
To fix visualizations that use prometheus heatmap format.

**Who is this feature for?**
Prometheus users

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/71017

**Special notes for your reviewer:**
This seems to get the vizs up and running and everything looks as it did before the breaking changes were merged. There's still more bugs that I've missed, but hopefully not anymore that are introduced?

This has been tested with and without the prometheusDataplane flag enabled, this should work in both cases.

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/result_transformer.ts [hunk 1]
```diff
@@ -352,7 +353,8 @@ export function transform(
 
   // When format is heatmap use the already created data frames and transform it more
   if (options.format === 'heatmap') {
-    return mergeHeatmapFrames(transformToHistogramOverTime(dataFrame.sort(sortSeriesByLabel)));
+    const sortedFrames = dataFrame.sort(sortSeriesByLabel);
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/result_transformer.ts:356]: revert this change

**Review by gtk-grafana — COMMENTED** [review]
Add unit tests. REJECTED

File: packages/grafana-data/src/types/dataFrame.ts [hunk 2]
```diff
@@ -266,6 +266,7 @@ export interface DataFrameDTO extends QueryResultBase {
 
 export interface FieldCalcs extends Record<string, any> {}
 
+/** @deprecated check data plane docs: https://grafana.github.io/dataplane **/
```
Comment (itsmylife) [inline packages/grafana-data/src/types/dataFrame.ts:269]:
```suggestion
/** @deprecated check data plane docs: https://grafana.github.io/dataplane/heatmap **/
```

**Review by itsmylife — APPROVED** [review]
LGTM I asked just a small change

**Pull request merged**